src/test/unit/scheduler/test_analysis.py

```diff
@@ -401,3 +405,44 @@ def test_is_forced_update(self):
         assert self.scheduler._is_forced_update(fo) is False
         fo.force_update = True
         assert self.scheduler._is_forced_update(fo) is True
+
+
+class TestAnalysisShouldReanalyse:
+    class PluginMock:
+        DEPENDENCIES = ['plugin_dep']
+        VERSION = '1.0'
+        NAME = 'plugin_root'
+
+    class BackendMock:
+        pass
+
+    @classmethod
+    def setup_class(cls):
+        cls.init_patch = mock.patch(target='scheduler.Analysis.AnalysisScheduler.__init__', new=lambda *_: None)
+        cls.init_patch.start()
+        cls.scheduler = AnalysisScheduler()
+        cls.init_patch.stop()
+
+    @pytest.mark.parametrize('plugin_root_date, plugin_dep_date, is_up_to_date', [
+        (10, 20, False),
+        (20, 10, True)
+    ])
+    def test_analysis_is_up_to_date(self, plugin_root_date, plugin_dep_date, is_up_to_date, monkeypatch):
+        def _analysis_get_date_mock(plugin_name, uid, backend_db_interface):
+            _ = backend_db_interface
+            _ = uid
```

Comment:
Did you do this to shut pylint up? :smile: 
Since the parameters are never used as keyword arguments, it should be fine to ignore them
```suggestion
        def _analysis_get_date_mock(plugin_name, *_):
```
or to use a `#pylint: disable=`